cli/src/pcluster/validators/s3_validators.py

```diff
@@ -0,0 +1,68 @@
+import re
+from urllib.error import HTTPError, URLError
+from urllib.parse import urlparse
+from urllib.request import urlopen
+
+from botocore.exceptions import ClientError
+
+from common.boto3.s3 import S3Client
+from pcluster.models.param import Param
+from pcluster.validators.common import FailureLevel, Validator
+
+
+class UrlValidator(Validator):
+    """Url Validator."""
+
+    def validate(self, url: Param):
+        """Validate given url with s3, https or file prefix."""
+        scheme = urlparse(url.value).scheme
+        if scheme in ["https", "s3", "file"]:
+            if scheme == "s3":
+                self._validate_s3_uri(url.value)
+            if scheme == "https":
+                try:
+                    urlopen(url.value)
+                except HTTPError as e:
+                    self._add_failure(
+                        "The url '{0}' cause HTTPError, the error code is '{1}', the error reason is '{2}'".format(
+                            url.value, e.code, e.reason
+                        ),
+                        FailureLevel.WARNING,
+                    )
+                except URLError as e:
+                    self._add_failure(
+                        "The url '{0}' causes URLError, the error reason is '{1}'".format(url.value, e.reason),
+                        FailureLevel.WARNING,
+                    )
+                except ValueError:
+                    self._add_failure(
+                        ("The value '{0}' is not a valid URL").format(url.value),
+                        FailureLevel.ERROR,
+                    )
+        else:
+            self._add_failure(
+                ("The value '{0}' is not a valid URL, " "choose URL with 'https', 's3' or 'file' prefix.").format(
```

Comment:
minor, there are etra quotes here: `, " "choose` probably due to autoformat.

Reply:
Removed